### PR TITLE
docs: update

### LINKED ISSUE #64: Documentation for Metadata & Visitors
PR #62 introduces the ability to create codec visitors. TODO: docs.

### PR DESCRIPTION
**Pull request by tjjfvi**
Fixes #64

### CODE REVIEW

File: common.ts [hunk 1]
```diff
@@ -268,17 +268,16 @@ export class CodecVisitor<R> {
   }
 
   /**
-   * Once Deno releases with TS 4.7, this can be used like so:
```
Comment (tjjfvi) [inline common.ts:271]: This happened

File: common.ts [hunk 2]
```diff
@@ -268,17 +268,16 @@ export class CodecVisitor<R> {
...
    * ```ts
-   * visitor.generic((visitor) => <T>() =>
+   * visitor.generic(<T>() =>
```
Comment (tjjfvi) [inline common.ts:272]: This is a bit cleaner IMO; didn't really like putting the old version into the docs

File: spread/codec.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
-import { Codec, createCodec } from "../common.ts";
+import { Codec, createCodec, Expand } from "../common.ts";
 
-export function spread<A, B>($a: Codec<A>, $b: Codec<B>): Codec<A & B> {
+export function spread<A, B>($a: Codec<A>, $b: Codec<B>): Codec<Expand<A & B>> {
```
Comment (tjjfvi) [inline spread/codec.ts:3]: Better intellisense

File: union/codec.ts [hunk 4]
```diff
@@ -48,7 +48,7 @@ export type NativeTaggedUnionMember<
 export type NativeTaggedUnionMembers<
   TK extends PropertyKey,
   M extends Record<number, TaggedUnionMember>,
-> = { [K in keyof M]: NativeTaggedUnionMember<TK, M[K & number]> }[number];
+> = { [K in keyof M]: NativeTaggedUnionMember<TK, Extract<M[K], TaggedUnionMember>> }[number];
```
Comment (tjjfvi) [inline union/codec.ts:51]: Fixed bug in typing

**Review by harrysolovay — APPROVED** [review]

**Pull request merged**